benchmarks/imagenet/resnet50/vicreg.py

```diff
@@ -0,0 +1,114 @@
+from typing import List, Tuple
+
+import torch
+from pytorch_lightning import LightningModule
+from torch import Tensor
+from torch.nn import Identity
+from torchvision.models import resnet50
+
+from lightly.loss.vicreg_loss import VICRegLoss
+from lightly.models.modules.heads import BarlowTwinsProjectionHead
+from lightly.models.utils import get_weight_decay_parameters
+from lightly.transforms.vicreg_transform import VICRegTransform
+from lightly.utils.benchmarking import OnlineLinearClassifier
+from lightly.utils.lars import LARS
+from lightly.utils.scheduler import CosineWarmupScheduler
+
+
+class VICReg(LightningModule):
+    def __init__(self, batch_size_per_device: int, num_classes: int) -> None:
+        super().__init__()
+        self.save_hyperparameters()
+        self.batch_size_per_device = batch_size_per_device
+
+        resnet = resnet50()
+        resnet.fc = Identity()  # Ignore classification head
+        self.backbone = resnet
+        # VICReg uses Barlow Twins projection head.
+        self.projection_head = BarlowTwinsProjectionHead()
+        self.criterion = VICRegLoss()
+
+        self.online_classifier = OnlineLinearClassifier(num_classes=num_classes)
+
+    def forward(self, x: Tensor) -> Tensor:
+        return self.backbone(x)
+
+    def training_step(
+        self, batch: Tuple[List[Tensor], Tensor, List[str]], batch_idx: int
+    ) -> Tensor:
+        views, targets = batch[0], batch[1]
+        features = self.forward(torch.cat(views)).flatten(start_dim=1)
+        z = self.projection_head(features)
+        z_a, z_b = z.chunk(len(views))
+        loss = self.criterion(z_a=z_a, z_b=z_b)
```

Comment:
In the simclr.py, it uses instead z0 and z1 instead. Is there a reason for this inconsistency?

```python
views, targets = batch[0], batch[1]
features = self.forward(torch.cat(views)).flatten(start_dim=1)
z = self.projection_head(features)
z0, z1 = z.chunk(len(views))
oss = self.criterion(z0, z1)
```

Reply:
Because the criterion expects `z_a` and `z_b` so I wanted to be consistent there.